### PR TITLE
Fix loadbalancer internal models to use TF types

### PR DESCRIPTION
**Pull request by hcsa73**

### CODE REVIEW

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 1]
```diff
@@ -642,31 +671,42 @@ func (r *loadBalancerResource) Update(ctx context.Context, req resource.UpdateRe
...
-	for _, targetPool := range model.TargetPools {
+	targetPools := []targetPool{}
+	diags = model.TargetPools.ElementsAs(ctx, &targetPools, false)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	for i := range targetPools {
+		targetPool := targetPools[i]
+		targetPoolName := model.Name.ValueString()
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:682]:
```suggestion
		targetPoolName := targetPool.Name.ValueString()
```

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 2]
```diff
@@ -642,31 +671,42 @@ func (r *loadBalancerResource) Update(ctx context.Context, req resource.UpdateRe
 	ctx = tflog.SetField(ctx, "project_id", projectId)
 	ctx = tflog.SetField(ctx, "name", name)
 
-	for _, targetPool := range model.TargetPools {
+	targetPools := []targetPool{}
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:674]: In some of the other fields, you call this variable  <field>Model but here it doesn't have the "Model" suffix

I would actually remove the "Model" suffix in the other variable names, seems redundant

**Review by vicentepinto98 — CHANGES_REQUESTED** [review]

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go (hunk 2 above)
Comment (hcsa73) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go:674]: I've renamed this variable to `targetPoolModel`, let me know if you want me to revert this and rename the other `...Model` variables

Comment (vicentepinto98) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go]: fine for me

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**